core/primitives/authority.hpp

```diff
@@ -7,17 +7,46 @@
 #define KAGOME_AUTHORITY_HPP
 
 #include <cstdint>
+#include <functional>
 
 #include "primitives/common.hpp"
 
 namespace kagome::primitives {
   using AuthorityWeight = uint64_t;
 
+  /**
+   * Authority id
+   */
+  struct AuthorityId {
+    SessionKey id;
+
+    bool operator==(const AuthorityId &other) const {
+      return id == other.id;
+    }
+    bool operator!=(const AuthorityId &other) const {
+      return !(*this == other);
+    };
```

Comment:
It is not sensible, because it makes impossible to use default constructor and  initialization list constructor and complicates initialization of item containers (used in tests)